crates/services/txpool_v2/src/selection_algorithms/ratio_tip_gas.rs

```diff
@@ -98,125 +124,106 @@ impl<S: RatioTipGasSelectionAlgorithmStorage> SelectionAlgorithm
 {
     type Storage = S;
     type StorageIndex = S::StorageIndex;
+
     fn gather_best_txs(
         &mut self,
         constraints: Constraints,
         storage: &S,
     ) -> Result<Vec<S::StorageIndex>, Error> {
         let mut gas_left = constraints.max_gas;
-        let mut best_transactions = Vec::new();
-
-        // Take the first transaction with the highest tip/gas ratio if it fits in the gas limit
-        // then promote all its dependents to the list of transactions to be executed
-        // and repeat the process until the gas limit is reached
+        let mut result = Vec::new();
+
+        // Take iterate over all transactions with the highest tip/gas ratio. If transaction
+        // fits in the gas limit select it and mark all its dependents to be promoted.
+        // Do that until end of the list or gas limit is reached. If gas limit is not
+        // reached, but we have promoted transactions we can start again from the beginning.
```

Comment:
This means that in reality, if the pool is quite used, there is a low chance that two dependents transactions are included in the same block ?

Reply:
Yes. It is not great, but at the same time in most cases we have executable transactions, not dependent. In this case the old algorithm will do a lot of iteration over the start of the list(in the case when start of the list doesn't fit requirements).

As I mentioned in the comments, it is still problem at the end of the list if promoted transactions now ca fit. But we need to address this problem in a separate PR

Reply:
Okey